source/dyaml/node.d

```diff
@@ -239,100 +237,115 @@ struct Node
          *                  a shortcut, like "!!int".
          */
         this(T)(T value, const string tag = null)
-            if(!scalarCtorNothrow!T && (!isArray!T && !isAssociativeArray!T))
         {
             tag_ = tag;
 
-            // No copyconstruction.
-            static assert(!is(Unqual!T == Node));
-
-            enum unexpectedType = "Unexpected type in the non-nothrow YAML node constructor";
-            static if(isSomeString!T)             { setValue(value.to!string); }
-            else static if(is(Unqual!T == Value)) { setValue(value); }
-            else static if(Value.allowed!T)       { static assert(false, unexpectedType); }
+            //Unlike with assignment, we're just copying the value.
+            static if (is(Unqual!T == Node)) {setValue(value.value_); }
```

Comment:
Can you put the {} on their own lines (+ indent) ?

Reply:
Yeah. I didn't want to mess with the style too much here, so I left it alone.